modules/eventlog/src/main/java/com/intuit/wasabi/eventlog/impl/EventLogImpl.java

```diff
@@ -180,13 +196,27 @@ public void run() {
      * @param event the vent to prepare
      */
     /*test*/ void prepareEnvelope(final EventLogEvent event) {
-        if (event == null) {
-            return;
-        }
-        for (EventLogListener eventLogListener : listeners.keySet()) {
-            if (isSubscribed(eventLogListener, event)) {
-                eventPostThreadPoolExecutor.submit(new EventLogEventEnvelope(event, eventLogListener));
+        LOGGER.debug("preparing event: {}", event);
+
+        if (event != null) {
```

Comment:
i prefer one exit per method

it is a bit much of the logging ... but each one of those helped in this case. will revisit.

defer